aws/resource_aws_codebuild_project.go

```diff
@@ -99,12 +99,18 @@ func resourceAwsCodeBuildProject() *schema.Resource {
 							ValidateFunc: validation.StringInSlice([]string{
 								codebuild.CacheTypeNoCache,
 								codebuild.CacheTypeS3,
+								codebuild.CacheTypeLocal,
 							}, false),
 						},
 						"location": {
 							Type:     schema.TypeString,
 							Optional: true,
 						},
+						"modes": {
+							Type:     schema.TypeList,
```

Comment:
If the API ordering of this attribute is not guaranteed when multiple modes are configured, we will likely need to change this to a `TypeSet`.